src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AttachmentController.cs

```diff
@@ -8,63 +9,125 @@ namespace TodoTemplate.Api.Controllers;
 public class AttachmentController : ControllerBase
 {
     private readonly AppSettings _appSettings;
+    private readonly UserManager<User> _userManager;
     private readonly IWebHostEnvironment _webHostEnvironment;
 
-    public AttachmentController(IOptionsSnapshot<AppSettings> setting, IWebHostEnvironment webHostEnvironment)
+    public AttachmentController(IOptionsSnapshot<AppSettings> setting,
+        UserManager<User> userManager,
+        IWebHostEnvironment webHostEnvironment)
     {
         _appSettings = setting.Value;
+        _userManager = userManager;
         _webHostEnvironment = webHostEnvironment;
     }
 
     [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
     [DisableRequestSizeLimit]
     [HttpPost("[action]")]
-    public async Task UploadProfilePhoto(IFormFile? file, CancellationToken cancellationToken)
+    public async Task UploadProfileImage(IFormFile? file, CancellationToken cancellationToken)
     {
         if (file is null)
             throw new BadRequestException();
 
-        await using var fileStream = file.OpenReadStream();
+        var userId = User.GetUserId();
 
-        Directory.CreateDirectory(_appSettings.UserProfilePhotoPath);
+        var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Id == userId, cancellationToken);
 
-        var userId = User.GetUserId();
+        if (user is null)
+            throw new ResourceNotFoundException(nameof(ErrorStrings.UserCouldNotBeFound));
 
-        var path = Path.Combine($"{_appSettings.UserProfilePhotoPath}\\{userId}{Path.GetExtension(file.FileName)}");
+        var profileImageName = Guid.NewGuid();
 
-        if (SystemFile.Exists(path))
-            SystemFile.Delete(path);
+        await using var fileStream = file.OpenReadStream();
+
+        Directory.CreateDirectory(_appSettings.UserProfileImagePath);
+
+        var path = Path.Combine($"{_appSettings.UserProfileImagePath}\\{profileImageName}{Path.GetExtension(file.FileName)}");
 
         await using var targetStream = SystemFile.Create(path);
+
         await fileStream.CopyToAsync(targetStream, cancellationToken);
+
+        if (user.ProfileImageName is not null)
+        {
+            try
+            {
+                var filePath = Directory.GetFiles(_appSettings.UserProfileImagePath,
+                    $"{user.ProfileImageName}.*").FirstOrDefault();
+
+                if (filePath != null)
+                {
+                    SystemFile.Delete(filePath);
+                }
+            }
+            catch
+            {
+                // not important
+            }
+        }
+
+        try
+        { 
+            user.ProfileImageName = profileImageName;
+
+            await _userManager.UpdateAsync(user);
+        }
+        catch
+        {
+            SystemFile.Delete(path);
+
+            throw new BadRequestException();
```

Comment:
plz just re-throw the exception here. do not introduce a new exception.
i meant use `throw;`